### ISSUE TITLE
[facebook-ios-sdk] - Different schemes building on master than in current release

### ISSUE THREAD

**Issue by nickmshelley** [Carthage/Carthage#2461]
* carthage install method: source
* `which carthage`: /usr/local/bin/carthage
* `carthage version`: 1295134ce2694aa60107398d2f83d136b3e86398
* `xcodebuild -version`: Xcode 9.3.1 Build version 9E501
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? No
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? No
* Are you using `--new-resolver`? No

**Cartfile**
```
github "facebook/facebook-ios-sdk"
```

**Carthage Output**
```
carthage update --platform iOS
*** Fetching facebook-ios-sdk
*** Fetching Bolts-ObjC
*** Checking out Bolts-ObjC at "1.9.0"
*** Checking out facebook-ios-sdk at "4.33.0"
*** xcodebuild output can be found in /var/folders/2l/ckm98g_j6x1_lwsmh_9nrby4_dk_j5/T/carthage-xcodebuild.1WkqYk.log
*** Downloading facebook-ios-sdk.framework binary at "4.33.0"
*** Building scheme "Bolts-iOS" in Bolts.xcodeproj
*** Building scheme "Bolts-iOS-Dynamic" in Bolts.xcodeproj
*** Building scheme "FBSDKCoreKit-Dynamic" in FacebookSDK.xcworkspace
*** Building scheme "FBSDKCoreKit-Universal" in FacebookSDK.xcworkspace
*** Building scheme "FBSDKLoginKit-Dynamic" in FacebookSDK.xcworkspace
*** Building scheme "FBSDKIntegrationTests" in FacebookSDK.xcworkspace
Build Failed
	Task failed with exit code 66:
	/usr/bin/xcrun xcodebuild -workspace /Users/nshelley/Desktop/TestApp/Carthage/Checkouts/facebook-ios-sdk/FacebookSDK.xcworkspace -scheme FBSDKIntegrationTests -configuration Release -derivedDataPath /Users/nshelley/Library/Caches/org.carthage.CarthageKit/DerivedData/9.3.1_9E501/facebook-ios-sdk/4.33.0 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/2l/ckm98g_j6x1_lwsmh_9nrby4_dk_j5/T/facebook-ios-sdk SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/nshelley/Desktop/TestApp/Carthage/Checkouts/facebook-ios-sdk)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/2l/ckm98g_j6x1_lwsmh_9nrby4_dk_j5/T/carthage-xcodebuild.1WkqYk.log
```

**Actual outcome**
Carthage tried to build `FBSDKIntegrationTests` scheme.

**Expected outcome**
It should not (as the current release does not)

**Comment by nickmshelley** [comment]
I narrowed the different behavior down to this commit: https://github.com/Carthage/Carthage/commit/70b797e4f22e5cf73a209e57479308d34d4f83e9#diff-f66644e8cfbcad3c3826395a690ddfbc.

So basically, Carthage didn't build static frameworks before, but now it does. And for some reason the `FBSDKIntegrationTests` scheme is coming through as a static framework.

This seems to me more like a Facebook bug more than a Carthage bug, so I'll close this.

Maybe I'll open an issue on their project. Oh wait, they disabled that...

**Comment by blender** [comment]
@nickmshelley I was also looking into this. I think the problem is 

`return frameworkType != nil`

FBSDKIntegrationTests has Mach-O type bundle which I don't know why is being turned into an allowed framework type

**Comment by chuganzy** [comment]
I also faced to this issue and made a PR: https://github.com/Carthage/Carthage/pull/2507

**Comment by blender** [comment]
Please try `brew install carthage --HEAD` #2507 has been merged and it should fix the issue

**Comment by ikesyo** [comment]
Addressed by #2507.
